lib/jekyll/static_file.rb

```diff
@@ -100,5 +101,40 @@ def to_liquid
         "path"          => File.join("", relative_path)
       }
     end
+
+    # Applies a similar URL-building technique as Jekyll::Document that takes
+    # the collection's URL template into account. The default URL template can
+    # be overriden in the collection's configuration in _config.yml.
+    def url
+      return @url unless @url.nil?
```

Comment:
Wow, had no idea that was possible... Done, though I did have to remove the `!` from `gsub!`, as that will return `nil` if no substitutions are performed.